src/services/allowlist.spec.ts

```diff
@@ -47,6 +49,15 @@ describe("validateCalldata", () => {
       expect(actual).toEqual({ success: false, error: "can't validate a tx that has no call data" });
     });
 
+    it("should return `true` and log warn message when contract address is `ZeroAddress`", async () => {
```

Comment:
Unsure if this is intended, added this test as it's what the code is doing right now

Reply:
Yeah, I changed the message in the error thrown to make more sense